dot/simswap/util/reverse2original.py

```diff
@@ -103,22 +98,21 @@ def reverse2wholeimage(
     use_mask=False,
     use_gpu=True,
 ):
-
     target_image_list = []
     img_mask_list = []
+    device = torch.device("cuda" if use_gpu else "cpu")
     if use_mask:
-        if use_gpu:
-            smooth_mask = SoftErosion(
-                kernel_size=17, threshold=0.9, iterations=7
-            ).cuda()
-        else:
-            smooth_mask = SoftErosion(kernel_size=17, threshold=0.9, iterations=7).cpu()
+        smooth_mask = SoftErosion(kernel_size=17, threshold=0.9, iterations=7).to(
+            device
+        )
     else:
         pass
 
     for swaped_img, mat, source_img in zip(swaped_imgs, mats, b_align_crop_tenor_list):
         swaped_img = swaped_img.cpu().detach().numpy().transpose((1, 2, 0))
-        img_white = np.full((crop_size, crop_size), 255, dtype=float)
+        img_white = np.full((crop_size, crop_size, 3), 255, dtype=np.uint8)
+        img_white = K.utils.image_to_tensor(img_white)
+        img_white = img_white[None, ...].float() / 255.0
 
         # inverse the Affine transformation matrix
         mat_rev = np.zeros([2, 3])
```

Comment:
These operation implement the explicit inversion of the matrix cell by cell, and they are done in numpy.
I think it's worth to try to move the to pytorch and call `inv` (https://pytorch.org/docs/stable/generated/torch.linalg.inv.html#torch.linalg.inv)

Reply:
Done https://github.com/sensity-ai/dot/blob/feature/3-speedup-simswap/dot/simswap/util/reverse2original.py#L136